src/Microsoft.AspNetCore.Http.Abstractions/HostString.cs

```diff
@@ -270,43 +327,43 @@ public override int GetHashCode()
         /// <summary>
         /// Parses the current value. IPv6 addresses will have brackets added if they are missing.
         /// </summary>
-        private void GetParts(out string host, out string port)
+        private static void GetParts(StringSegment value, out StringSegment host, out StringSegment port)
         {
             int index;
             port = null;
             host = null;
 
-            if (string.IsNullOrEmpty(_value))
+            if (StringSegment.IsNullOrEmpty(value))
             {
                 return;
             }
-            else if ((index = _value.IndexOf(']')) >= 0)
+            else if ((index = value.IndexOf(']')) >= 0)
             {
                 // IPv6 in brackets [::1], maybe with port
-                host = _value.Substring(0, index + 1);
+                host = value.Subsegment(0, index + 1);
 
-                if ((index = _value.IndexOf(':', index + 1)) >= 0)
+                if ((index = value.IndexOf(':', index + 1)) >= 0)
                 {
-                    port = _value.Substring(index + 1);
+                    port = value.Subsegment(index + 1);
                 }
             }
-            else if ((index = _value.IndexOf(':')) >= 0
-                && index < _value.Length - 1
-                && _value.IndexOf(':', index + 1) >= 0)
+            else if ((index = value.IndexOf(':')) >= 0
+                && index < value.Length - 1
+                && value.IndexOf(':', index + 1) >= 0)
             {
                 // IPv6 without brackets ::1 is the only type of host with 2 or more colons
-                host = $"[{_value}]";
+                host = $"[{value}]";
                 port = null;
             }
             else if (index >= 0)
             {
                 // Has a port
-                host = _value.Substring(0, index);
-                port = _value.Substring(index + 1);
+                host = value.Subsegment(0, index);
+                port = value.Subsegment(index + 1);
```

Comment:
Do you think we should verify the port is actually a number? Having a public MatchesAny API that strips ports that might not actually be ports seems fragile.

Reply:
The format is validated elsewhere.

Reply:
That would be a more compelling argument if this wasn't called as part of a public API that takes arbitrary strings and doesn't validate the port is a number. Unless you're saying MatchesAny does this somewhere I'm not seeing.

Reply:
And if it's not numeric, then what? Include it in the host?

Reply:
Either that or throw a FormatException.